epochStart/bootstrap/storageProcess.go

```diff
@@ -72,7 +72,11 @@ func NewStorageEpochStartBootstrap(args ArgsStorageEpochStartBootstrap) (*storag
 
 	esbConverted, ok := esb.(*epochStartBootstrap)
 	if !ok {
-		return nil, errors.ErrInvalidTypeConversion
+		sovereignESB, ok := esb.(*sovereignChainEpochStartBootstrap)
```

Comment:
I think we should go for the clean version and use factories from runTypeComponents actually.

Reply:
+1

Reply:
refactored